### ISSUE TITLE
Let pydoctor know about links to zope.interface

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8515]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#8515|
|Type|enhancement|
|Created|2016-06-27 17:44:35Z|
|Branch|https://github.com/twisted/twisted/tree/8515-pydoctor-zope-interface|

This was raised as part of [#8513](https://github.com/twisted/twisted/issues/8513)

zope.interface docs are at https://docs.zope.org/zope.interface

<details><summary>Searchable metadata</summary>

```
trac-id__8515 8515
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__None None
branch__8515_pydoctor_zope_interface 8515-pydoctor-zope-interface
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1467049475146502 1467049475146502
changetime__1467731298219221 1467731298219221
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

Here is the PR https://github.com/twisted/twisted/pull/238

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban removed owner|
|-|-|

I have no idea how to reference the zope.interface.Interface and zope.interface.adapter.AdapterRegistry so I converted them to C{}

You can see the build history for this branch with various attempts 

I can see that the zope interface apidoc is here https://docs.zope.org/zope.interface/api.html ... and is a bit of a chaos

Please check that all is ok.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban commented|
|-|-|

I have pushed a new code in which a custom pydoctor system is used and for which `zope.interface.adapter.AdapterRegistry` is handled as a special case with a manual URL resolution.

With this we should be able to use the valid L{} marker in our code.

Once upstream documentation is fixed we can remove it from the custom pydoctor system.

As a drive by I have also fixed all zope.interface references in the code.

The current APIBuilderTests tests are broken but they should be fixed in [#8518](https://github.com/twisted/twisted/issues/8518)

For this ticket I will add tests for the new twisted.python._pydoctor module.

For now, please check that you are ok with the direction of the attached branch.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban commented|
|-|-|

We need to fix [#8518](https://github.com/twisted/twisted/issues/8518) before merging this so that we can update the tests

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_5"></a>|@rodrigc commented|
|-|-|

See my comments:
https://github.com/twisted/twisted/pull/238#issuecomment-229265548

I think this is OK to commit, once you solve the other issues.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_6"></a>|@rodrigc set owner to @adiroiban|
|-|-|

Looks OK.  Please get this in so we can reduce errors from the epydoc build.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_7"></a>|@adiroiban commented|
|-|-|

Thanks for the review! It was not looking OK ... in fact it was pretty bad :)

The TwistedModuleVisitor was never really used and the sphinx hack has no test coverage... the test_pydoctor.py is empty.

Also, the code was not documented.

I am working on this and will send back for review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban removed owner|
|-|-|

Ok. The branch should be much better now.

Since pydoctor is not installed for testing, the tests are skipped but you can run the tests on your local machine.

I am working on  [#8518](https://github.com/twisted/twisted/issues/8518) to enable pydoctor tests and fix the apibuilder tests that were broken since we don't execute them.

Please review!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_9"></a>|@rodrigc set owner to @adiroiban|
|-|-|

OK

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_10"></a>|GitHub set status to `closed`|
|-|-|

In changeset 2af5ff26647d4237c5267a6d41d70b1a6d9537d9
```
#!CommitTicketReference repository="" revision="2af5ff26647d4237c5267a6d41d70b1a6d9537d9"
Merge 8515-pydoctor-zope-interface: Add intersphinx links to zope.interface.

Author: adiroiban
Reviewer: rodrigc
Fixes: #8515
```
